src/riscv/lib/src/machine_state/page_cache/state.rs

```diff
@@ -50,12 +51,12 @@ struct PageEntry {
 /// connection with the concrete types that implement these traits.
 ///
 /// [`PageCache`]: super::PageCache
-pub struct PageCacheImpl<const PAGES: usize> {
-    pages: Box<[Option<PageEntry>; PAGES]>,
+pub struct PageCacheImpl<const PAGES: usize, CPE: CodePageEntry> {
+    pages: Box<[Option<PageEntry<CPE>>; PAGES]>,
```

Comment:
This is a little off-topic, so feel free to resolve.

The `pages` field expresses a nesting `box -> array -> option -> page entry -> boxed -> array -> ...`. I wonder if the `option -> page entry -> boxed -> ...` portion could be expressed as `option -> boxed -> page entry -> ...` instead to give the Rust compiler a little help to choose the optimal representation for the `Option`, given `Option<Box<...>>` is ideal thanks to the non-nullness of `Box`.

In other words, if you move the `Box` from `PageEntry` into the `Option` in `PageCacheImpl`.

```suggestion
    pages: Box<[Option<Box<PageEntry<CPE>>>; PAGES]>,
```

I don't know if this speeds up things, or if it even saves on dereferences. It only exploits the special representations the compiler picks for `Option<Box<_>>`, which it might already do.

Reply:
I've raised [RV-790](https://linear.app/tezos/issue/RV-790/page-cache-investigate-changing-optionpageboxentries-to) for now to look into this properly later. Doing it now does make the initialisation of `PageEntry` a little more complicated.